### PR TITLE
mariadb: update to 10.5.9.

### PR DESCRIPTION
**Pull request by Veganizer**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64 x86_64-musl)
- [x] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [x] aarch64-musl
  - [x] armv7l
  - [x] armv6l-musl

Cross compiling works now!

[ci skip]

### CODE REVIEW

**Comment by ericonr** [comment]
Might make it easier to help to share the relevant parts of 

```
2021-04-11T04:31:52.2089340Z See also "/builddir/mariadb-10.5.9/build/CMakeFiles/CMakeOutput.log".
2021-04-11T04:31:52.2090601Z See also "/builddir/mariadb-10.5.9/build/CMakeFiles/CMakeError.log".
```

**Comment by Veganizer** [comment]
```bash
Determining if the function floor exists failed with the following output:
Change Dir: /builddir/mariadb-10.5.9/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_d867c && [1/2] Building C object CMakeFiles/cmTC_d867c.dir/CheckFunctionExists.c.o
<command-line>: warning: conflicting types for built-in function 'floor'; expected 'double(double)' [-Wbuiltin-declaration-mismatch]
/usr/share/cmake-3.19/Modules/CheckFunctionExists.c:7:3: note: in expansion of macro 'CHECK_FUNCTION_EXISTS'
    7 |   CHECK_FUNCTION_EXISTS(void);
      |   ^~~~~~~~~~~~~~~~~~~~~
/usr/share/cmake-3.19/Modules/CheckFunctionExists.c:1:1: note: 'floor' is declared in header '<math.h>'
  +++ |+#include <math.h>
    1 | #ifdef CHECK_FUNCTION_EXISTS
[2/2] Linking C executable cmTC_d867c
FAILED: cmTC_d867c 
: && /builddir/.xbps-mariadb/wrappers/aarch64-linux-gnu-gcc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -march=armv8-a   -I/usr/aarch64-linux-gnu/usr/include -pie -fPIC -fstack-protector --param=ssp-buffer-size=4 -moutline-atomics -DCHECK_FUNCTION_EXISTS=floor -Wl,-z,relro -Wl,-z,now -Wl,--as-needed    -L/usr/aarch64-linux-gnu/usr/lib   -rdynamic CMakeFiles/cmTC_d867c.dir/CheckFunctionExists.c.o -o cmTC_d867c   && :
/usr/lib/gcc/aarch64-linux-gnu/10.2.1/../../../../aarch64-linux-gnu/bin/ld: CMakeFiles/cmTC_d867c.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text.startup+0x14): undefined reference to `floor'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
This for example.

File: srcpkgs/mariadb/template [hunk 1]
```diff
@@ -72,8 +49,14 @@ post_install() {
 	rm -f ${DESTDIR}/usr/share/man/man1/mysql-test-run.pl.1
 	rm -f ${DESTDIR}/usr/bin/mytop
 
-	# Configuration file.
-	install -Dm644 ${DESTDIR}/usr/share/mysql/my-medium.cnf ${DESTDIR}/etc/mysql/my.cnf
+	mkdir -p ${DESTDIR}/usr/lib/security
```
Comment (Johnnynator) [inline srcpkgs/mariadb/template:52]:
```suggestion
	vmkdir usr/lib/security
```

**Comment by Veganizer** [comment]
fixed crosscompiling now. :-)

**Comment by Johnnynator** [comment]
```
CMake Deprecation Warning at cmake/modules/SetKDbCMakePolicies.cmake:12 (cmake_policy):
  The OLD behavior for policy CMP0059 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  CMakeLists.txt:4 (include)


-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /builddir/.xbps-kdb/wrappers/aarch64-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /builddir/.xbps-kdb/wrappers/aarch64-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at cmake/modules/KDbAddSimpleOption.cmake:29 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'BUILD_COVERAGE'.
Call Stack (most recent call first):
  cmake/modules/KDbAddTests.cmake:35 (simple_option)
  CMakeLists.txt:10 (kdb_add_tests)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Installing in the same prefix as Qt, adopting their path scheme.
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Performing Test HAVE_DATE_TIME
-- Performing Test HAVE_DATE_TIME - Success
-- Found KF5CoreAddons: /usr/aarch64-linux-gnu/usr/lib64/cmake/KF5CoreAddons/KF5CoreAddonsConfig.cmake (found version "5.81.0") 
-- Found KF5: success (found suitable version "5.81.0", minimum required is "5.16.0") found components: CoreAddons 
-- Found the following ICU libraries:
--   i18n (required)
--   uc (required)
-- Found ICU: /usr/aarch64-linux-gnu/usr/include (found version "67.1") 
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Found PythonInterp: /usr/aarch64-linux-gnu/usr/bin/python2 (Required is at least version "2.6") 
-- Creating shared data class in KDbConnectionData.h from KDbConnectionData.shared.h
-- Creating shared data class in KDbObject.h from KDbObject.shared.h
-- Creating shared data class in KDbQuerySchemaParameter.h from KDbQuerySchemaParameter.shared.h
-- Creating shared data class in KDbResult.h from KDbResult.shared.h
-- Creating shared data class in KDbSelectStatementOptions.h from KDbSelectStatementOptions.shared.h
-- Creating shared data class in KDbVersionInfo.h from KDbVersionInfo.shared.h
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
CMake Error at cmake/modules/KDbGenerateHeaders.cmake:170 (message):
  Could not find "/builddir/kdb-3.2.0/build/src/KDbConnectionData.h"
Call Stack (most recent call first):
  src/CMakeLists.txt:402 (ecm_generate_headers)
```
It seems like kdb-3.2.0 fails to build for aarch64 cross (and maybe more)

File: srcpkgs/mariadb/files/musl-disable-failing-test.patch [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+ SET(ma_pagecache_single_src     ma_pagecache_single.c test_file.c test_file.h)
+ SET(ma_pagecache_consist_src ma_pagecache_consist.c test_file.c test_file.h)
+--- ./unittest/mysys/CMakeLists.txt	2020-11-10 14:15:06.000000000 +0100
++++ ./unittest/mysys/CMakeLists.txt	2020-11-24 21:08:46.721183004 +0100
+@@ -15,7 +15,7 @@
+
+ MY_ADD_TESTS(bitmap base64 my_atomic my_rdtsc lf my_malloc my_getopt dynstring
+              byte_order
+-             queues stacktrace crc32 LINK_LIBRARIES mysys)
```
Comment (grooverdan) [inline srcpkgs/mariadb/files/musl-disable-failing-test.patch:24]: MDEV-24131 was fixed in 10.5.9

**Review by Johnnynator — APPROVED** [review]

**Pull request merged**